src/jvm/main/org/jetbrains/kotlinx/lincheck/Lincheck.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck
+
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionResult
+import org.jetbrains.kotlinx.lincheck.execution.ExecutionScenario
+import org.jetbrains.kotlinx.lincheck.execution.parallelResults
+import org.jetbrains.kotlinx.lincheck.strategy.LincheckFailure
+import org.jetbrains.kotlinx.lincheck.strategy.managed.forClasses
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck.strategy.verify
+import org.jetbrains.kotlinx.lincheck.transformation.withLincheckJavaAgent
+import org.jetbrains.kotlinx.lincheck.verifier.Verifier
+import kotlin.reflect.jvm.javaMethod
+
+
+object Lincheck {
+    @JvmStatic
+    fun <R> verifyWithModelChecker(
+        invocations: Int = 1,
+        block: () -> R
+    ): LincheckFailure? {
+        val scenario = ExecutionScenario(
+            emptyList(),
+            listOf(
+                listOf(
+                    Actor(
+                        Wrapper<R>::run.javaMethod!!,
+                        listOf(block)
+                    )
+                )
+            ),
+            emptyList(),
+            null
+        )
+
+        val options = ModelCheckingOptions()
+            .iterations(0)
+            .invocationsPerIteration(invocations)
+            .addCustomScenario(scenario)
+            .addGuarantee(forClasses(Lincheck::class).allMethods().ignore())
+            .addGuarantee(forClasses(Wrapper::class).allMethods().ignore())
+            .verifier(ExecutionExceptionsVerifier::class.java)
+
+        val testCfg = options.createTestConfigurations(Wrapper::class.java)
+
+        withLincheckJavaAgent(testCfg.instrumentationMode) {
+            val strategy = testCfg.createStrategy(Wrapper::class.java, scenario, null, null)
+            val verifier = testCfg.createVerifier()
+
```

Comment:
BTW, I think somewhere here you need to enforce the bytecode transformation of the `block::class.java` (i.e. the class of the passed lambda). Otherwise the events inside the lambda would not be tracked. 

I think once you'll add the tests you will see the problem.